pat/ReferenceTestV2/reference_consumer_v2.py

```diff
@@ -184,648 +112,63 @@ def _on_description_modification_report(self, received_message_data: ReceivedMes
-        my_ops = [op for op in operations if op.Type.Code == '67108890']
-        if len(my_ops) != 1:
-            results_collector.log_result(False, step, info, f'found {len(my_ops)} operations with code "67108890"')
-        else:
-            operation = my_ops[0]
-            proposed_metric_state1 = client.mdib.xtra.mk_proposed_state('numeric_metric_0.channel_0.vmd_1.mds_0')
-            proposed_metric_state2 = client.mdib.xtra.mk_proposed_state('numeric_metric_1.channel_0.vmd_1.mds_0')
-            for st in (proposed_metric_state1, proposed_metric_state2):
-                if st.MetricValue is None:
-                    st.mk_metric_value()
-                    st.MetricValue.Value = Decimal(1)
-                else:
-                    st.MetricValue.Value += Decimal('0.1')
-            future_object = client.set_service_client.set_metric_state(
-                operation.Handle,
-                [proposed_metric_state1, proposed_metric_state2],
-            )
-            operation_result = future_object.result()
-            if len(operation_result.report_parts) == 0:
-                results_collector.log_result(False, step, info, 'no notification')
-            elif len(operation_result.report_parts) > 1:
-                results_collector.log_result(
-                    False,
-                    step,
-                    info,
-                    f'got {len(operation_result.report_parts)} notifications, expect only one',
-                )
-            else:
-                results_collector.log_result(
-                    True,
-                    step,
-                    info,
-                    f'got {len(operation_result.report_parts)} notifications',
-                )
-            if operation_result.InvocationInfo.InvocationState != msg_types.InvocationState.FINISHED:
-                results_collector.log_result(
-                    False,
-                    step,
-                    info,
-                    f'got result {operation_result.InvocationInfo.InvocationState} '
-                    f'{operation_result.InvocationInfo.InvocationError} '
-                    f'{operation_result.InvocationInfo.InvocationErrorMessage}',
-                )
-    except Exception as ex:  # noqa: BLE001
-        print(traceback.format_exc())
-        results_collector.log_result(False, step, info, ex)
-
-    step = '7'
-    info = 'Graceful shutdown (at least subscriptions are ended; optionally Bye is sent)'
-    try:
-        success = client._subscription_mgr.unsubscribe_all()  # noqa: SLF001
-        results_collector.log_result(success, step, info)
-    except Exception as ex:  # noqa: BLE001
-        print(traceback.format_exc())
-        results_collector.log_result(False, step, info, ex)
-    time.sleep(2)
-    return results_collector
+    if not step_1.test_1b(wsd, search_epr):
+        return
+    services = wsd.search_services(timeout=-1)

```

Comment:
would suggest to set a proper timeout here, otherwise it might search forever without any info/log. Or add some more log messages

Reply:
modify WSDiscovery to have a methode returning the last discovered services